NExtends/Expressions/ExpressionMerger.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Linq.Expressions;
+
+namespace NExtends.Expressions
+{
+	public class ExpressionMerger : ExpressionVisitor
+	{
+		Expression CurrentParameterExpression { get; set; }
+
+		public static Expression<Func<TIn, TOut>> Merge<TIn, TA, TOut>(Expression<Func<TIn, TA>> entryPoint, Expression<Func<TA, TOut>> expression1)
+		{
+			return new ExpressionMerger().MergeAll<TIn, TA, TOut>(entryPoint, expression1);
+		}
+
+		public static Expression<Func<TIn, TOut>> Merge<TIn, TA, TB, TOut>(Expression<Func<TIn, TA>> entryPoint, Expression<Func<TA, TB>> expression1, Expression<Func<TB, TOut>> expression2)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of generic parameters in the 'ExpressionMerger.Merge' method to no more than the 2 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS2436)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Consider using more general type 'System.Linq.Expressions.LambdaExpression' instead of 'System.Linq.Expressions.Expression<System.Func<TA, TB>>'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS3242)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Consider using more general type 'System.Linq.Expressions.LambdaExpression' instead of 'System.Linq.Expressions.Expression<System.Func<TB, TOut>>'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS3242)